src/ocean/meta/types/ReduceType.d

```diff
@@ -0,0 +1,199 @@
+/*******************************************************************************
+
+    Generic utility allowing to recursively visit an arbitrary type and reduce
+    its definition to some compile-time value. It is intended to be used as
+    implementation cornerstone for complex type traits to address recursive
```

Comment:
A bit clearer: "...implementation cornerstone for complex type traits to avoid having to rewrite the recursive type reflection boilerplate wherever it's needed."